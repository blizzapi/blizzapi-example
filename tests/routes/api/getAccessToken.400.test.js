const request = require('supertest');
const app = require('../../../src/app');

jest.mock('blizzapi', () => jest.fn().mockImplementation(
  () => ({
    getAccessToken: (region) => {
      throw new Error('wrong response');
    }
  }),
));

describe('/getAccessToken/:region (error 400)', () => {
  const wrongRoute = '/api/getAccessToken/wrong';

  test('should send status code 400 in case of error', async () => {
    const response = await request(app).get(wrongRoute);
    expect(response.statusCode).toBe(400);
  });

  test('error response should match snapshot', async () => {
    const response = await request(app).get(wrongRoute);
    expect(response.body).toMatchSnapshot();
  });
});