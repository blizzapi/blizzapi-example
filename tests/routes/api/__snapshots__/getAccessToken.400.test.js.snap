// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/getAccessToken/:region (error 400) error response should match snapshot 1`] = `
Object {
  "error": "Failed to receive Battle.net API data",
}
`;
